### PR TITLE
Nav block: extract hook for inner blocks

### PR DESCRIPTION
**Pull request by getdave**
## What?

Extracts a new hook for handling inner blocks related things.

Also removes unrelated concept of "submenus" from scope of the exported data.

## Why?

Having this within the body of the `edit.js` file was making it very verbose and noisy. This PR reduces noisy and is part of an effort to make the block more approachable by codify concepts.

## How?

Extracts a custom hook for inner blocks related things.

## Testing Instructions

Basically check the block doesn't break in any major way.

Especially worth checking that submenus still work as expected.

### CODE REVIEW

File: packages/block-library/src/navigation/edit/index.js [hunk 1]
```diff
@@ -194,37 +193,13 @@ function Navigation( {
...
-				isInnerBlockSelected: hasSelectedInnerBlock( clientId, true ),
-			};
-		},
-		[ clientId ]
+		innerBlocks,
+	} = useInnerBlocks( clientId );
+
+	const hasSubmenus = !! innerBlocks.find(
+		( block ) => block.name === 'core/navigation-submenu'
 	);
```
Comment (getdave) [inline packages/block-library/src/navigation/edit/index.js:197]: This concept of "submenus" was previously baked into the `useSelect` which handled information about inner blocks.

I find the concept to be unrelated to the extent that it should not be included in the hook.

**Review by scruffian — APPROVED** [review]
LGTM

**Comment by getdave** [comment]
Waiting on tests...
